src/WSServerCon.cpp

```diff
@@ -596,6 +596,19 @@ void WSServerCon::processMessage(const QString &message)
         }
         );
     }
+    else if (root["msg"] == "lock_device")
+    {
+        mpdevice->lockDevice([this, root](bool success, QString errstr)
+        {
+            if (!success) return;
```

Comment:
if !success you need to send back sendFailedJson(root, errstr); before returning.

Reply:
The point was the if it works then it's evident by the fact that the device's status changes to locked. But sure, can send back response in every case.

Reply:
Haha.. I actually meant `if (success) return`..